external/CMakeLists.txt

```diff
@@ -247,19 +249,25 @@ target_include_directories(vma INTERFACE
 	"${CMAKE_CURRENT_SOURCE_DIR}/VulkanMemoryAllocator-Hpp/include"
 	"${CMAKE_CURRENT_SOURCE_DIR}/VulkanMemoryAllocator-Hpp/VulkanMemoryAllocator/include")
 
-add_subdirectory(ffmpeg)
+if (MINGW)
+	add_library(ffmpeg INTERFACE)
+	target_include_directories(ffmpeg INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/ffmpeg/include")
+	target_link_libraries(ffmpeg INTERFACE avformat avcodec swscale avutil avfilter swresample)
```

Comment:
To be honest, that seems like a pretty bad idea. Vita3K is currently built using ffmpeg 6 while msys2 has already updated to ffmpeg 7. Moreover, Vita3K is using ffmpeg's internal API, which means using a different minor version (or possibly even patch) compared to the headers and can cause a runtime crash.
Wouldn't it be possible to add a mingw-compatible version to the ffmpeg subdirectory?

Reply:
I understand what you are saying, but currently I am the only one who is building using mingw-w64 and you do not distribute binaries built with mingw-w64 anyway. So I'd rather not invest a lot of time into something that only I would use anyway. We could always do it later, currently it does not build on mingw-w64 anyway without my changes.

Reply:
I mean the goal to contribute and make a PR for an open-source project is for other people to be able to use it. If you are the only one able to use the content of your PR, it's really hard to approve it this way.

About ffmpeg, the ffmpeg submodule automatically generate binaries using vcpkg. Vcpkg has support for mingw (x64-mingw-static triplet). Assuming it compiles correctly with it (it's not guaranteed given that it's a community triplet), adding support to it would only require a few lines in the CI file.

Reply:
I reverted this change. I tested building ffmpeg with `x64-mingw-static` and it worked. Do you want a PR in the https://github.com/Vita3K/ffmpeg-core repo?

Reply:
Yes please, given that only prebuilts used for the current target are downloaded when using ffmpeg, I see no issue in adding another triplet.